include/cuco/bloom_filter_policy.cuh

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuco/detail/bloom_filter/bloom_filter_policy_impl.cuh>
+#include <cuco/hash_functions.cuh>
+
+#include <cuda/std/array>
+
+#include <cstdint>
+
+namespace cuco {
+
+// TODO Policies are meant as customization points so we need to ensure a user-defined policy
+// fulfills a certain concept.
+
+/**
+ * @brief A policy that defines how a Blocked Bloom Filter generates and stores a key's fingerprint.
+ *
+ * @note `Block` is used **only** to determine `block_words` via `cuda::std::tuple_size<Block>` and
+ * `word_type` via `Block::value_type` and does not represent the actual storage type of the filter.
+ * We recommend using `cuda::std::array`.
+ *
+ * @tparam Hash Hash function used to generate a key's fingerprint
+ * @tparam Block Type to determine the filter's block size and underlying word type
+ */
+template <class Hash, class Block>
+class bloom_filter_policy {
+  using impl_type = cuco::detail::bloom_filter_policy_impl<Hash, Block>;
+
+ public:
+  using hasher             = typename impl_type::hasher;              ///< Type of the hash function
+  using hash_argument_type = typename impl_type::hash_argument_type;  ///< Hash function input type
+  using hash_result_type   = typename impl_type::hash_result_type;    ///< hash function output type
+  using word_type =
+    typename impl_type::word_type;  ///< Underlying machine word type of the filter's storage
+
+  static constexpr std::uint32_t words_per_block =
+    impl_type::words_per_block;  ///< Number of machine words in each filter block
+
+ public:
+  /**
+   * @brief Constructs the `bloom_filter_policy`object.
+   *
+   * @note This policy dynamically adjusts the `pattern_bits` to set at least one bit per word in
```

Comment:
WRT "This policy": I think the fact that is is **_a_** policy, not _**the**_ policy argues for renaming the class. The name `bloom_filter_policy` could be reserved for the concept, once C++20 arrives to this repo.

Reply:
Agreed. I renamed the class to `default_filter_policy`.